docs/snippets/composition-environments/composition.yaml

```diff
@@ -0,0 +1,52 @@
+---
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: xvpcs.aws.platformref.upbound.io
+  labels:
+    provider: aws
+spec:
+  environment:
+    environmentConfigs:
+      - ref:
+          name: example-environment
+  compositeTypeRef:
+    apiVersion: aws.platformref.upbound.io/v1alpha1
+    kind: XVPC
+  resources:
+    - name: vpc
+      base:
+        apiVersion: ec2.aws.crossplane.io/v1beta1
+        kind: VPC
+        spec:
+          forProvider:
+            cidrBlock: 172.16.0.0/16
+      patches:
+        - fromFieldPath: spec.vpcName
+          toFieldPath: spec.forProvider.tags.Name
```

Comment:
If we are using comminity provider we need to change patch style https://marketplace.upbound.io/providers/crossplane-contrib/provider-aws/v0.33.0/resources/ec2.aws.crossplane.io/VPC/v1beta1#doc:spec-forProvider-tags

Otherwise it fails with
```
  Warning  ComposeResources   4m3s                    defined/compositeresourcedefinition.apiextensions.crossplane.io  cannot render composed resource from resource template at index 0: cannot use dry-run create to name composed resource: VPC.ec2.aws.crossplane.io "vpc-using-environment-config-wzxtw" is invalid: spec.forProvider.tags: Invalid value: "object": spec.forProvider.tags in body must be of type array: "object"
```

The current form will work with the official provider though as we have `tags` as `object` there https://marketplace.upbound.io/providers/upbound/provider-aws/v0.24.0/resources/ec2.aws.upbound.io/VPC/v1beta1#doc:spec-forProvider-tags

Reply:
Thanks for the hint. I switched to the community provider and changed `tag` to an array.